File: JavaScript/58_LengthOfLastWord.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLastWord = function(s) {
    let i = s.length - 1;
    let count = 0;
    while (s[i] === " ") {
        i--;
    }

    while (i >= 0 && s[i] !== " ") {
        count ++;
        i--;
    }

    return count;
};


const testCases = [
    {
        s: "Hello World",
        expected: 5
    },
    {
        s: "   fly me   to   the moon  ",
        expected: 4
    },
    {
        s: "luffy is still joyboy",
        expected: 6
    },
]

testCases.forEach((testCase, i) => {
    const output = lengthOfLastWord(testCase.s);
    if (JSON.stringify(output) === JSON.stringify(testCase.expected)) {
        console.log(`Test Case ${i}: Passed. Expected: ${JSON.stringify(testCase.expected)}, Output: ${JSON.stringify(output)} `)
    } else {
        console.log(`Test Case ${i}: Failed. Expected: ${JSON.stringify(testCase.expected)}, Output: ${JSON.stringify(output)} `)
    }
});

File: CS/58_LengthOfLastWord.cs
namespace LeetCodeSolutions {
    public static class LengthOfLastWord {
        public static int Solution(string s) {
            int i = s.Length - 1;
            int count = 0;
            while (s[i] == ' ') {
                i--;
            }
            while (i >= 0 && s[i] != ' ') {
                i--;
                count++;
            }
            return count;
        }
        public static void RunTests() {
            Console.WriteLine("Running Length Of Last Word Tests...");

            var testCases = new List<(string, int)> {
                ("Hello World", 5),
                ("   fly me   to   the moon  ", 4),
                ("luffy is still joyboy", 6),
            };

            foreach (var (s, expected) in testCases) {
                try {
                    var result = Solution(s);
                    Console.WriteLine(
                        $"Input: {s} " +
                        $"Expected: {expected} Result: {result}"
                    );
                } catch (Exception ex) {
                    Console.WriteLine($"Test Failed: {ex.Message}");
                }
            }
        }
    }
}